projects/plugins/jetpack/modules/comments/comments.php

```diff
@@ -493,65 +493,66 @@ public function watch_comment_parent() {
+						comm_par = ( comm_par_el && comm_par_el.value ) ? comm_par_el.value : '',
+						frame = document.getElementById( 'jetpack_remote_comment' ),
+						tellFrameNewParent;
+
+					tellFrameNewParent = function () {
+						if ( comm_par ) {
+							frame.src = "<?php echo esc_url_raw( $this->signed_url ); ?>" + '&replytocom=' + parseInt( comm_par, 10 ).toString();
+						} else {
+							frame.src = "<?php echo esc_url_raw( $this->signed_url ); ?>";
+						}
+					};
 
-				<?php if ( get_option( 'thread_comments' ) && get_option( 'thread_comments_depth' ) ) : ?>
+					<?php if ( get_option( 'thread_comments' ) && get_option( 'thread_comments_depth' ) ) : ?>
 
-				if ( 'undefined' !== typeof addComment ) {
-					addComment._Jetpack_moveForm = addComment.moveForm;
+					// Check addComment._Jetpack_moveForm to make sure we don't monkey-patch twice.
+					if ( 'undefined' !== typeof addComment && ! addComment._Jetpack_moveForm ) {
+						addComment._Jetpack_moveForm = addComment.moveForm;
 
-					addComment.moveForm = function ( commId, parentId, respondId, postId ) {
-						var returnValue = addComment._Jetpack_moveForm( commId, parentId, respondId, postId ),
-							cancelClick, cancel;
+						addComment.moveForm = function ( commId, parentId, respondId, postId ) {
+							var returnValue = addComment._Jetpack_moveForm( commId, parentId, respondId, postId ),
+								cancelClick, cancel;
 
-						if ( false === returnValue ) {
-							cancel = document.getElementById( 'cancel-comment-reply-link' );
-							cancelClick = cancel.onclick;
-							cancel.onclick = function () {
-								var cancelReturn = cancelClick.call( this );
-								if ( false !== cancelReturn ) {
-									return cancelReturn;
-								}
+							if ( false === returnValue ) {
+								cancel = document.getElementById( 'cancel-comment-reply-link' );
+								cancelClick = cancel.onclick;
+								cancel.onclick = function () {
+									var cancelReturn = cancelClick.call( this );
+									if ( false !== cancelReturn ) {
+										return cancelReturn;
+									}
 
-								if ( ! comm_par ) {
-									return cancelReturn;
-								}
+									if ( ! comm_par ) {
+										return cancelReturn;
+									}
 
-								comm_par = 0;
+									comm_par = 0;
 
-								tellFrameNewParent();
+									tellFrameNewParent();
 
-								return cancelReturn;
-							};
-						}
+									return cancelReturn;
+								};
+							}
 
-						if ( comm_par == parentId ) {
-							return returnValue;
-						}
+							if ( comm_par == parentId ) {
+								return returnValue;
+							}
 
-						comm_par = parentId;
+							comm_par = parentId;
 
-						tellFrameNewParent();
+							tellFrameNewParent();
 
-						return returnValue;
-					};
-				}
+							return returnValue;
+						};
+					}
 
-				<?php endif; ?>
+					<?php endif; ?>
 
-				// Do the post message bit after the dom has loaded.
-				document.addEventListener( 'DOMContentLoaded', function () {
```

Comment:
Should we be concerned that the event listener is going to be registered twice now, once from `DOMContentLoaded` and once from the script's `load`?

For that matter, it looks like we don't need any of the variables except `frame` in the case that `thread_comments` are disabled. So possibly we should restructure all this something more like this:
```
var frame = document.getElementById( 'jetpack_remote_comment' );

<?php if ( get_option( 'thread_comments' ) && get_option( 'thread_comments_depth' ) ) : ?>

function watchReply() {
	// Check addComment._Jetpack_moveForm to make sure we can monkey-patch but don't do so twice.
	if ( 'undefined' === typeof addComment || addComment._Jetpack_moveForm ) {
		return;
	}

	... All the stuff above except setting `frame` ...
}

document.addEventListener( 'DOMContentLoaded', watchReply );
// in WP 6.4+, the script is loaded asynchronously, so we need to wait for it to load before we monkey-patch the functions it introduces.
document.querySelector( '#comment-reply-js' )?.addEventListener( 'load', watchReply );

<?php endif; ?>

// Do the post message bit after the dom has loaded.
document.addEventListener( 'DOMContentLoaded', function () {
	... the stuff that was below ...
} );
```